src/modules/service-provider/api/provider.controller.spec.ts

```diff
@@ -46,14 +45,14 @@ describe('Provider Controller Test', () => {
                 },
             ],
         })
-            .overrideProvider(RolleRepo)
-            .useValue(createMock<RolleRepo>())
+            .overrideProvider(ServiceProviderService)
+            .useValue(createMock<ServiceProviderService>())
             .overrideProvider(ServiceProviderRepo)
             .useValue(createMock<ServiceProviderRepo>())
             .compile();
 
-        serviceProviderRepoMock = module.get(ServiceProviderRepo);
-        rolleRepoMock = module.get(RolleRepo);
+        serviceProviderServiceMock = await module.resolve<DeepMocked<ServiceProviderService>>(ServiceProviderService);
+        serviceProviderRepoMock = await module.resolve<DeepMocked<ServiceProviderRepo>>(ServiceProviderRepo);
```

Comment:
Any reason for the switch to `.resolve`? These aren't scoped providers, are they?
If the only reason is "they are injected in a different module, so they are not accessible" you could use `.get(xxx, {strict: false})`

Reply:
See my comment about the circular dependency. I think this was left over from experimentation.

Fixed in [5596065](https://github.com/dBildungsplattform/dbildungs-iam-server/pull/648/commits/55960656bfa01f7dbb6cd2368742605f7e14b81c)